File: Level1/220825/JavaScript/MockTest.test.js
import {
  solution, firstCountCompute, secoundCountCompute, thirdCountCompute,
} from './MockTest';

test('firstCount', () => {
  expect(firstCountCompute(0, [1, 3, 2, 4, 2])).toBe(2);
});

test('secondCount', () => {
  expect(secoundCountCompute(0, [1, 2, 4, 1, 2])).toBe(1);
});

test('thirdCount', () => {
  expect(thirdCountCompute(0, [3, 1, 1, 2, 4, 4, 4])).toBe(3);
});

test('solution', () => {
  expect(solution([1, 2, 3, 4, 5])).toStrictEqual([1]);
  expect(solution([1, 3, 2, 4, 2])).toStrictEqual([1, 2, 3]);
});

File: Programmers/Level1/220824/Java/MockTestTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static org.junit.jupiter.api.Assertions.assertEquals;

class MockTestTest {
  @Test
  void firstCount() {
    MockTest mockTest = new MockTest();

    assertEquals(5, mockTest.gradingFirstAnswers(
        new int[]{1, 2, 3, 4, 5}, new int[]{1, 2, 3, 4, 5}));
  }

  @Test
  void secondCount() {
    MockTest mockTest = new MockTest();

    assertEquals(2, mockTest.gradingSecondAnswers(
        new int[]{1, 3, 2, 4, 2}, new int[]{2, 1, 2, 3, 2, 4, 2, 5}));
  }

  @Test
  void thirdCount() {
    MockTest mockTest = new MockTest();

    assertEquals(2, mockTest.gradingThirdAnswers(
        new int[]{1, 3, 2, 4, 2}, new int[]{3, 3, 1, 1, 2, 2, 4, 4, 5, 5}));
  }

  @Test
  void answer() {
    MockTest mockTest = new MockTest();

    assertArrayEquals(new int[]{1},
        mockTest.solution(new int[]{1, 2, 3, 4, 5}));

    assertArrayEquals(new int[]{1, 2, 3},
        mockTest.solution(new int[]{1, 3, 2, 4, 2}));
  }
}
